### PR TITLE
fix(cli): output to stdout not working with multiple output formatters

### LINKED ISSUE #2042: Output to stdout not working with multiple output formatters
**Describe the bug**

spectral lint fails with error message `The number of outputs must match the number of formats` when on the command line there are two formatters specified but only one output file

**To Reproduce**
1. Given an arbitrary OpenAPI document `my-api.yaml` and any ruleset `.spectral.yaml`
2. Run this CLI command: `spectral lint --format stylish --format junit --output.junit junit-results.xml my-api.yaml`
3. See error  `[usage message omitted] The number of outputs must match the number of formats`

**Expected behavior**

junit formatted output is written to junit-results.xml while stylish formatted output is written to stdout.

Expecting this behavior since the usage message says
` -o, --output                   where to output results, can be a single file name, multiple "output.<format>" or missing to print to stdout`

**Environment**
 - spectral version: 6.2.0
 - OS: Microsoft Windows 10 Enterprise Version 10.0.19042 Build 19042

**Additional context**
No error when formatter count matches output file count, e.g. with
`spectral lint --format stylish --format junit --output.stylish stylish-results --output.junit junit-results.xml my-api.yml`

### PR DESCRIPTION
**Pull request by P0lip**
Fixes #2042 

**Checklist**

- [x] Tests added / updated
- [x] Docs added / updated

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/cli/package.json [hunk 1]
```diff
@@ -54,11 +54,11 @@
     "strip-ansi": "6.0",
     "text-table": "0.2",
     "tslib": "^2.3.0",
-    "yargs": "17.2.1"
+    "yargs": "17.3.1"
```
Comment (marbemac) [inline packages/cli/package.json:57]: Was this a regression with yargs or something?

File: packages/cli/src/commands/__tests__/lint.test.ts [hunk 2]
```diff
@@ -204,7 +207,7 @@ describe('lint', () => {
   });
 
   it('shows help if unknown format is passed', () => {
```
Comment (marbemac) [inline packages/cli/src/commands/__tests__/lint.test.ts:209]: Perhaps pop a test in here to verify something like `spectral lint --format stylish --format junit --output.junit junit-results.xml my-api.yaml` doesn't regress in future?

**Review by marbemac — CHANGES_REQUESTED** [review]

File: packages/cli/package.json (hunk 1 above)
Comment (P0lip) [reply packages/cli/package.json:57]: nope, it wasn't, I just updated it while I was at it.

File: packages/cli/src/commands/__tests__/lint.test.ts (hunk 2 above)
Comment (P0lip) [reply packages/cli/src/commands/__tests__/lint.test.ts:209]: we have it already haha :D
See https://github.com/stoplightio/spectral/issues/2042#issuecomment-1026193383

When I made the following change https://github.com/stoplightio/spectral/pull/2044/files#diff-5c1bab693e4315f93ddc3bc536502fe23b536bbed6214ca5a093571db6389f24R14-R21, the test started to fail.

**Comment by P0lip** [comment]
there are no actual changes, I just tested this new beauty 🤓 
image

**Review by marbemac — APPROVED** [review]
💯

**Pull request merged**